services/service-helpers/validator.js

```diff
@@ -23,6 +23,73 @@ v.validate = function(schema, config) {
   });
 };
 
+v.oneOf = function(options) {
+  if (!Array.isArray(options)) {
+    throw new Error('oneOf expects an array');
+  }
+  var checks = [];
+  options.forEach(function(option) {
+    if (!option || !option.__check) {
+      throw new Error(option + ' is not a valid option');
+    }
+    checks.push(option.__check);
+  });
+
+  return wrapCheck(function(value) {
+    var messages = checks
+      .map(function(check) {
+        return check(value);
+      })
+      .filter(function(message) {
+        return message;
+      });
+
+    // no match
+    if (messages.length === checks.length) {
+      return messages.join(' or ');
```

Comment:
The resultant message is a little bit awkward: `{something} must be a string or {something} must be a number`. A more natural sentence would be `{something} must be a string or a number`. Is there a way we can rearrange where & how `must be` is inserted to allow for that?

Reply:
I am assuming that all messages must follow the convention `must be {XYZ}`. Which allows me to remove all of the `must be` to possibly fix the problem. Thoughts ?

Reply:
Yes, I think they should all be in the form of `must be {a|an} {thing}`.